pkg/harvester/components/SettingList.vue

```diff
@@ -50,10 +57,61 @@ export default {
       handler() {
         this['categorySettings'] = this.filterCategorySettings();
       }
+    },
+    searchQuery: {
+      immediate: true,
+      handler(newQuery) {
+        this.filteredSettings = this.filterSearchSettings(this.categorySettings, newQuery);
+      }
     }
   },
 
   methods: {
+    filterSearchSettings(settings, searchKey) {
+      if (!searchKey) {
+        return this.filterCategorySettings();
+      }
+      const searchQuery = searchKey.toLowerCase();
+
+      return settings.filter((setting) => {
+        const id = setting.id?.toLowerCase() || '';
+
+        // filter by id
+        if (id.includes(searchQuery) ) {
+          return true;
+        }
+
+        const description = this.t(setting.description, {}, true)?.toLowerCase() || '';
+
+        // filter by description
+        if (description.includes(searchQuery)) {
+          return true;
+        }
+
+        // filter by customized value
+        if (setting.customized === true && setting.data?.value) {
+          const value = setting.data.value?.toLowerCase() || '';
+
+          return value.includes(searchQuery);
+        }
+
+        // filter by json value
+        if (setting.kind === 'json' && setting.json) {
+          // clean up the json string to remove unwanted characters
+          const jsonValue = setting.json?.toLowerCase()?.replace('(\n|\/|\{|\"|\})/gm', '') || '';
```

Comment:
`.replace('(\n|\/|\{|\"|\})/gm', '')` won’t work because the pattern is a string, not a RegExp, use `/(\n|\/|\{|\"|\})/gm` instead.

Reply:
[Nit] It might be more reliable to parse the JSON and search recursively.